Feature/#32 add form nickname

torahoo opened the pull request:
게시글 상세페이지에 nickname 보이게끔 수정
Post Entity 및 각각의 DTO 추가
이에 맞게끔 service와 controller 변경

torahoo committed:
feature/#32-addFormNickname :
1. List에 로그인 정보 뿌리기
2. addForm에 로그인 정보 보이기
3. postPage에 글쓴이 보이게끔 하기
4. header.html에 게시글 리스트(PostList) 버튼 추가

torahoo committed:
feature/#32-addFormNickname :
1. PostEntity 수정
2. AddPostDTO 작성
3. EditPostDTO 작성
4. PageDTO
5. ListDTO
6. schema.sql 최신화
7. 폴더 구성 변경 (domain.post->domain.post.service)
8. Entity 및 DTO 구성에 맞게 Service, Controller 변경

godchaean reviewed (commented):
확인

godchaean reviewed (approved):
확인

Hoogokok commented on geulkkoli/src/main/java/com/geulkkoli/web/post/dto/EditDTO.java:
이미 적절한 생성자가 선언되어 있어서 굳이 setter를 쓰지 않아도 @ModelAttribute시 객체 바인딩이 될거 같아요.
https://minchul-son.tistory.com/546 를 확인해봐도 좋을 거 같네요

takoeats commented on geulkkoli/src/main/java/com/geulkkoli/web/post/PostController.java:
```diff
@@ -17,59 +25,76 @@ public class PostController {
-    public String postAddForm(Model model) {
-        model.addAttribute("post", new Post());
+    public String postAddForm(Model model, HttpServletRequest request) {
+        model.addAttribute("post", new AddDTO());
```
빈 객체?

seungin2 replied:
thymeleaf를 이용한 HTML 자동 완성을 위함입니다. 
th:field="객체.멤버필드"를 HTML 태그에 입력 시 멤버필드와 같은 이름으로 id와 name을 자동으로 추가해줍니다.
김영한 강의에 나온 방법으로 꼭 필요하지는 않습니다.

Hoogokok commented on geulkkoli/src/main/java/com/geulkkoli/web/post/PostController.java:
User loginUser = (User) session.getAttribute(SessionConst.LOGIN_USER);

if(Objects.isNull(loginUser)){
           return "redirect:/post/list";
}

로 바꾸면 좋을 거 가탕요

torahoo committed:
feature/#32-addFormNickname :
1. Controller에 add부분 git review의견 반영
2. AddForm.html 세션 닉네임 받아오는 것을 controller쪽으로 변경
3. TestDataInit test유저 추가

takoeats reviewed (approved):
check

seungin2 reviewed (approved):
merge
